### DISCUSSION TITLE
Multiple instances of TFT_eSPI to work with two displays

### DISCUSSION THREAD

**Discussion by Florin196** [Bodmer/TFT_eSPI#3033] [Q&A - General]
Hello everyone!

I have a quick question on how to initiate two instances of TFT_eSPI to work with two simiar displays (GC9A01). Is there any way to set the two TFT_CSs from my Arduino ino file for each instance of the TFT_eSPI? 

I manage to have my code working controlling the CSs with digialWrite() but this is not a good solution given that I use LVGL and I defined two separate displays.

Thanks!

**Comment by Bodmer** [comment]
The library does not support multiple instances. The screen update in LVGL is just a simple buffer copy so options are:
1. manage the CS line in the sketch
2. Processors are so cheap that one per screen may be an option
3. Create a new library with just buffer transfer

**Comment by Florin196** [comment]
Many thanks for your prompt reply and good suggestions. I will look into option 3.Under option 1, I was not to able to have two lvgl animations running in parallel. The project will included in a larger system running on a SoC and option 2 is not possible.Kind regards,FlorinOn Dec 12, 2023, at 5:09 AM, Bodmer ***@***.***> wrote:﻿
The library does not support multiple instances. The screen update in LVGL is just a simple buffer copy so options are:

manage the CS line in the sketch
Processors are so cheap that one per screen may be an option
Create a new library with just buffer transfer
